### PR TITLE
Onboard azdo build pipeline

### PR DESCRIPTION
**Pull request by erdembayar**
Start adding Azdo pipeline for building artifacts internally.
Signing step not enabled, it looks M2 approval for that.

`azure-pipelines-official.yml` - official pipeline.
`azure-pipelines`- public build pipeline.

### CODE REVIEW

File: azure-pipelines-official.yml [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+    # arcade templates depend on the name.'
+    - task: PublishBuildArtifacts@1
+      displayName: Publish Packages
+      inputs:
+        PathtoPublish: '$(Build.SourcesDirectory)\artifacts\packages\$(BuildConfiguration)'
+        ArtifactName: 'PackageArtifacts'
+      condition: succeeded()
+
+    # Publish Asset Manifests for Build Asset Registry job
+    - task: PublishBuildArtifacts@1
```
Comment (nkolev92) [inline azure-pipelines-official.yml]: I can't imagine we need BAR here right?

**Review by nkolev92 — COMMENTED** [review]
Did you copy a template from somewhere?

File: azure-pipelines-official.yml (hunk 1 above)
Comment (erdembayar) [reply azure-pipelines-official.yml]: I copied from `dotnet-format`.
Sure, removed.

File: azure-pipelines-official.yml [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+      displayName: Publish MicroBuild Artifacts
+      inputs:
+        PathtoPublish: '$(Build.ArtifactStagingDirectory)\MicroBuild\Output'
+        ArtifactName: MicroBuildOutputs
+        publishLocation: Container
+      condition: succeededOrFailed()
+
+  # Publish to Build Asset Registry
+  - template: /eng/common/templates/job/publish-build-assets.yml
```
Comment (erdembayar) [inline azure-pipelines-official.yml]: @nkolev92 
Should remove this one too?

Comment (nkolev92) [reply azure-pipelines-official.yml]: Yep.

Comment (erdembayar) [reply azure-pipelines-official.yml]: Removed.

File: azure-pipelines-official.yml [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+    #             /p:DotNetSymbolServerTokenMsdl=$(microsoft-symbol-server-pat)
+    #             /p:DotNetSymbolServerTokenSymWeb=$(symweb-symbol-server-pat)
+    #             /p:DotNetFinalVersionKind=$(_additionalBuildArgs)
+    #             /p:DotNetArtifactsCategory=$(_DotNetArtifactsCategory)
+    #             /p:DotnetPublishUsingPipelines=true
+    #   displayName: Build
+
+    - template: eng\common\templates\steps\generate-sbom.yml
+
+    - task: PublishTestResults@2
```
Comment (nkolev92) [inline azure-pipelines-official.yml]: We're trying to publish the results, but I don't think a build has run yet.

Comment (erdembayar) [reply azure-pipelines-official.yml]: Sure, removed. I'm still learning yml logic.

**Review by nkolev92 — APPROVED** [review]

**Pull request merged**